pymc/gp/hsgp.py

```diff
@@ -0,0 +1,357 @@
+        """
+
+        # Index Xs using input_dim and active_dims of covariance function
+        Xs, _ = self.cov_func._slice(Xs)
+
+        # If not provided, use Xs and c to set L
+        if self._L is None:
+            assert isinstance(self._c, (float, np.ndarray, pt.TensorVariable))
+            self.L = set_boundary(Xs, self._c)
+        else:
+            self.L = self._L
+
+        eigvals = calc_eigenvalues(self.L, self._m, tl=pt)
+        phi = calc_eigenvectors(Xs, self.L, eigvals, self._m, tl=pt)
+        omega = pt.sqrt(eigvals)
+        psd = self.cov_func.power_spectral_density(omega)
+
+        i = int(self._drop_first == True)
+        return phi[:, i:], pt.sqrt(psd[i:])
+
+    def prior(self, name: str, X: TensorVariable, *args, **kwargs):
+        R"""
+        Returns the (approximate) GP prior distribution evaluated over the input locations `X`.
+
+        Parameters
+        ----------
+        name: string
+            Name of the random variable
+        X: array-like
+            Function input values.
+        dims: None
+            Dimension name for the GP random variable.
+        """
+        self._X_mean = pt.mean(X, axis=0)
+        phi, sqrt_psd = self.prior_linearized(X - self._X_mean)
+
+        if self._parameterization == "noncentered":
+            self._beta = pm.Normal(
+                f"{name}_hsgp_coeffs_", size=self._m_star - int(self._drop_first)
+            )
+            self._sqrt_psd = sqrt_psd
+            f = self.mean_func(X) + phi @ (self._beta * self._sqrt_psd)
+
+        elif self._parameterization == "centered":
+            self._beta = pm.Normal(f"{name}_hsgp_coeffs_", sigma=sqrt_psd)
+            f = self.mean_func(X) + phi @ self._beta
+
+        self.f = pm.Deterministic(name, f, dims=kwargs.get("dims"))
+        return self.f
+
+    def _build_conditional(self, Xnew):
+        try:
+            beta, X_mean = self._beta, self._X_mean
+
+            if self._parameterization == "noncentered":
+                sqrt_psd = self._sqrt_psd
+
+        except AttributeError:
+            raise ValueError(
+                "Prior is not set, can't create a conditional.  Call `.prior(name, X)` first."
+            )
+
+        Xnew, _ = self.cov_func._slice(Xnew)
+        eigvals = calc_eigenvalues(self.L, self._m, tl=pt)
+        phi = calc_eigenvectors(Xnew - X_mean, self.L, eigvals, self._m, tl=pt)
+        i = int(self._drop_first == True)
+
+        if self._parameterization == "noncentered":
+            return self.mean_func(Xnew) + phi[:, i:] @ (beta * sqrt_psd)
+
+        elif self._parameterization == "centered":
+            return self.mean_func(Xnew) + phi[:, i:] @ beta
+
+    def conditional(self, name: str, Xnew: TensorVariable, *args, **kwargs):
+        R"""
+        Returns the (approximate) conditional distribution evaluated over new input locations
+        `Xnew`.  If using the
+
+        Parameters
+        ----------
+        name: string
```

Comment:
```suggestion
        name
```
Should be `name : str` or since there's already a type hint in the signature IMO it's okay to leave it out here